### ISSUE TITLE
Gradle ignores customizations

### ISSUE THREAD

**Issue by przemek-baranowski** [akaita/easylauncher-gradle-plugin#10]
I've used this config, and plugin still generates only default green ribbon.

```groovy
apply plugin: 'com.akaita.android.easylauncher'

...

easylauncher {
    iconNames "@ mipmap/ic_launcher"  // there is no space after @
    foregroundIconNames "@ mipmap/ic_launcher_foreground" 
    productFlavors {
        useTest {
            filters = orangeRibbonFilter()
        }

        useProd {
            filters = redRibbonFilter()
        }

        withoutAppsee {

        }

        withAppsee {
            filters = overlayFilter(new File("app/launcherOverlay/appsee.png"))
        }
    }
}
```

**Comment by akaita** [comment]
Note: excuse me for editing your post. I had to add some quotes to make it more readable.

Could you please share your list of `flavors` and `types`?

**Comment by przemek-baranowski** [comment]
thanks for edit ;) All of flavours are repeated in the snippet above (4 flavours in 2 dimensions, 2 in each dimension), besides that i have 2 build types: debug and release.

**Comment by akaita** [comment]
I see. I wanted to confirm whether `useProd` was a flavor or a variant. It is a flavor then...

I don't see anything wrong, except for the spaces after the `@` (btw: I don't really know why you'd put those spaces there).

I'm going to have to ask a bunch of questions here:
 - Which file did you put the `easylauncher` config into?
 - What does `android:icon` in the manifest say?
 - You say you get the default green ribbon. For which flavors do you get it?

**Comment by akaita** [comment]
It is odd. This will require replicability.

Could you share a stripped down version of your project? 
I mean, something with just one empty activity, no resources, a different launcher icon if you want to, ...

**Comment by przemek-baranowski** [comment]
tiny update - plugis sees my config. I've noticed that defaultFlavorNaming param works

**Comment by przemek-baranowski** [comment]
I would rather not share my code, sorry. Anyway, I've migrated to ribbonizer , and config like below works like a charm.

`ribbonizer {
    builder { variant, iconFile ->
        if (variant.flavorName.contains("withAppsee")) {
            def filter = yellowRibbonFilter(variant, iconFile)
            filter.label = "APPSEE" + variant.versionCode
            return filter
        } else if (variant.flavorName.contains("useTest")) {
            def filter = yellowRibbonFilter(variant, iconFile)
            filter.label = "TEST"
            return filter
        } else if (variant.flavorName.contains("useProd")) {
            def filter = greenRibbonFilter(variant, iconFile)
            filter.label = "PROD"
            return filter
        }

        return null
    }
}`

**Comment by akaita** [comment]
I'm happy to see it worked out for you.

I'd still like to know what was going on. Could you print out the value of `variant.flavorName`?

**Comment by przemek-baranowski** [comment]
my example build variant name is `useProdWithAppseeDebug` so flavour name must be `useProdWithAppsee` which is combination of 2 flavours: `useProd` and `withAppsee`. Did you handled scenario for more than one flavour dimensions?

**Comment by akaita** [comment]
I see, that is the key then.  
I did not really try with multiple dimensions.

Right now this would work then:
```groovy
easylauncher {
    [...]
    productFlavors {
        useProdWithAppsee {
            filters = orangeRibbonFilter()
        }
        [...]
    }
}
```


I will think about supporting multiple dimensions, although it will probably not happen. I think once it reaches the plugin dimensions don't exist anymore, only flavors and types... I created #13 to keep track of it.
Thanks for sharing!
